> > @charlienegri : I add here again the time-chunking with `chunks={"time": 24}`. We tried that with CAMS2_83 and time=1 and while it gave good improvements on memory-consumption, it took slightly longer. With time=24, I couldn't see any performance degration, and the memory reduction of daily hourly data vs yearly hourly data is still huge. So, you might want to try that approach again?
> 
> I have installed the branch in the test module and will try a long run with it if it gets merged it will anyway be deployed to production in july as part of a new version of the module

test run crashed immediately with
```
Loading cams2_83-evaluation/test
  Loading requirement: proj/9.1.0
Traceback (most recent call last):
  File "/modules/rhel8/user-apps/fou-modules/cams2_83-evaluation/test/venv/bin/cams2_83", line 5, in <module>
    from pyaerocom.scripts.cams2_83.cli import app
  File "/modules/rhel8/user-apps/fou-modules/cams2_83-evaluation/test/venv/lib/python3.10/site-packages/pyaerocom/__init__.py", line 9, in <module>
    from .config import Config
  File "/modules/rhel8/user-apps/fou-modules/cams2_83-evaluation/test/venv/lib/python3.10/site-packages/pyaerocom/config.py", line 19, in <module>
    from pyaerocom.grid_io import GridIO
  File "/modules/rhel8/user-apps/fou-modules/cams2_83-evaluation/test/venv/lib/python3.10/site-packages/pyaerocom/grid_io.py", line 2, in <module>
    from pyaerocom.time_config import TS_TYPES
  File "/modules/rhel8/user-apps/fou-modules/cams2_83-evaluation/test/venv/lib/python3.10/site-packages/pyaerocom/time_config.py", line 7, in <module>
    from iris import coord_categorisation
  File "/modules/rhel8/user-apps/fou-modules/cams2_83-evaluation/test/venv/lib/python3.10/site-packages/iris/coord_categorisation.py", line 23, in <module>
    import iris.coords
  File "/modules/rhel8/user-apps/fou-modules/cams2_83-evaluation/test/venv/lib/python3.10/site-packages/iris/coords.py", line 23, in <module>
    from iris.common import (
  File "/modules/rhel8/user-apps/fou-modules/cams2_83-evaluation/test/venv/lib/python3.10/site-packages/iris/common/__init__.py", line 9, in <module>
    from .mixin import *
  File "/modules/rhel8/user-apps/fou-modules/cams2_83-evaluation/test/venv/lib/python3.10/site-packages/iris/common/mixin.py", line 10, in <module>
    import cf_units
  File "/modules/rhel8/user-apps/fou-modules/cams2_83-evaluation/test/venv/lib/python3.10/site-packages/cf_units/__init__.py", line 23, in <module>
    from cf_units import _udunits2 as _ud
  File "cf_units/_udunits2.pyx", line 1, in init cf_units._udunits2
ValueError: numpy.dtype size changed, may indicate binary incompatibility. Expected 96 from C header, got 88 from PyObject
```
I will try a fresh venv

--- a/pyaerocom/io/cams2_83/reader.py
+++ b/pyaerocom/io/cams2_83/reader.py
@@ -194,6 +194,7 @@ def read_dataset(paths: list[Path], *, day: int) -> xr.Dataset:
         paths,
         preprocess=preprocess,
         parallel=False,
+        chunks={"time": 24}
     )
     return ds.pipe(fix_coord).pipe(fix_names)
 

--- a/pyaerocom/io/cams2_83/reader.py
+++ b/pyaerocom/io/cams2_83/reader.py
@@ -190,12 +190,7 @@ def read_dataset(paths: list[Path], *, day: int) -> xr.Dataset:
     def preprocess(ds: xr.Dataset) -> xr.Dataset:
         return ds.pipe(forecast_day, day=day).pipe(fix_missing_vars)
 
-    ds = xr.open_mfdataset(
-        paths,
-        preprocess=preprocess,
-        parallel=False,
-        chunks={"time": 24}
-    )
+    ds = xr.open_mfdataset(paths, preprocess=preprocess, parallel=False, chunks={"time": 24})
     return ds.pipe(fix_coord).pipe(fix_names)
 
 

